### PR TITLE
WIP: Autocomplete

### PR DESCRIPTION
**Pull request by vjousse**
Tracked here #44

### CODE REVIEW

File: src/Model.elm [hunk 1]
```diff
@@ -123,6 +138,11 @@ type alias Model =
     , useGlobalTimeline : Bool
     , viewer : Maybe Viewer
     , currentView : CurrentView
+    , autocompleteState : Autocomplete.State
+    , autocompleteAccounts : List Mastodon.Model.Account
+    , autocompleteQuery : String
```
Comment (n1k0) [inline src/Model.elm]: How about putting these in a dedicated `AutocompleteInfo` typed field?

File: src/View.elm [hunk 2]
```diff
@@ -717,6 +723,25 @@ viewerView { attachments, attachment } =
             ]
 
 
+
+-- setup for your autocomplete view
```
Comment (n1k0) [inline src/View.elm]: ?

File: src/Model.elm [hunk 3]
```diff
@@ -336,60 +382,98 @@ deleteStatusFromTimeline statusId timeline =
...
+    in
+        case draftMsg of
+            ClearDraft ->
+                { model | draft = defaultDraft } ! []
+
+            ToggleSpoiler enabled ->
+                { model
+                    | draft =
+                        { oldDraft
+                            | spoiler_text =
```
Comment (n1k0) [inline src/Model.elm]: Nit: This is worth putting in a `let` statement.

**Comment by vjousse** [comment]
To be continued on #107

**Pull request closed without merging**